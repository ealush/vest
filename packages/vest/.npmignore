src
playground
config
testUtils
./jest.config.js
dist