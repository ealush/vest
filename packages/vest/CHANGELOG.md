# vest - Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## 1.0.31 - 2021-07-02
### Fixed and improved
- e6ea7d6  support global replaces (undefined)
- 34e0414  improved conditions (undefined)
- 26c28c6  all tests pass (undefined)
- 33f4e46  release (undefined)
- 6fe40c7  better bundle (undefined)
- c2cfb65  better typing (undefined)
- c6387ab  before ts settings (undefined)
- c0e9708  generate correct d.ts file (undefined)
- packages/vest/types/vest.development.d.ts
- 8e01b8e  x (undefined)
- afb3960  x (undefined)
- e0a8463  add changelog support (undefined)
- cc46c38  current (undefined)
- 4a2073f  patch(vest): changes (ealush)
- c0a1705  patch(vest): changes (ealush)
- 240386c  better create generic (ealush)
- b6db1c6  transform any to unknowns (ealush)
- f2c20a7  add done integration tests (ealush)
- ba56e33  produce methods (ealush)
- 0638243  better produce integration tests (ealush)
- 4098a11  getFailures as integration tests (ealush)
- f56cf7f  pd (ealush)
- 2ea440e  pd (ealush)
- b3032ef  relocate draftresult (ealush)
- 962141f  test getFailuresByGroup (ealush)
- e163c56  test getFailures (ealush)
- a92b6df  test collectFailureMessages (ealush)
- 81aad51  fix most tests (ealush)
- c149e90  simplify collectFailureMessages (ealush)
