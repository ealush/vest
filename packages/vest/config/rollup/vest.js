const addEsmDir = require('../../../../config/rollup/addEsmDir');
const genConfig = require('../../../../config/rollup/genConfig');
const writeMainTemplate = require('../../../../config/rollup/writeMainTemplate');
const { packageDist, packageNames } = require('../../../../util');

addEsmDir(packageDist(packageNames.VEST));
writeMainTemplate(packageDist(packageNames.VEST), packageNames.VEST);

export default genConfig({
  distPath: packageDist(packageNames.VEST),
  esInput: 'vest.mjs.js',
  input: 'vest.js',
  libraryName: packageNames.VEST,
});
