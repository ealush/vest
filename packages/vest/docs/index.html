<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <title>Vest - Validation Test</title>
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
    <meta name="description" content="Validation Test" />
    <meta
      name="viewport"
      content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
    />
    <link rel="icon" href="./_assets/favicon.ico" />
    <link
      rel="stylesheet"
      href="https://cdn.jsdelivr.net/npm/docsify-themeable@0/dist/css/theme-simple.css"
    />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/prismjs@1.22.0/themes/prism-tomorrow.css">
  </head>
  <body>
    <div id="app"></div>
    <script>
      window.$docsify = {
        name: 'vest',
        logo: '/_assets/vest-logo.png',
        repo: 'https://github.com/ealush/vest',
        search: 'auto',
        loadSidebar: true,
        subMaxLevel: 2,
        themeColor: '#5BA9D8',
      };
    </script>
    <script src="//unpkg.com/docsify/lib/docsify.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/docsify-themeable@0"></script>
    <script src="//unpkg.com/docsify/lib/plugins/search.min.js"></script>
    <script src="//unpkg.com/vest"></script>
    <script src="./_assets/index.js"></script>
  </body>
</html>
