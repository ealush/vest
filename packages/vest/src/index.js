import any from "anyone/any";
import enforce from "n4s/dist/enforce.min";
import { VERSION } from "./constants";
import test from "./core/test";
import validate from "./core/validate";
import { draft, only, skip, warn } from "./hooks";
import { singleton, runWithContext } from "./lib";

export default singleton.register({
  Enforce: enforce.Enforce,
  VERSION,
  any,
  draft,
  enforce,
  only,
  runWithContext,
  skip,
  test,
  validate,
  warn,
});
