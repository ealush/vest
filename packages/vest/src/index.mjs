import enforce from '../../n4s/src/enforce';
import create from './core/suite/create';
import test from './core/test';
import { draft, only, skip, warn, group } from './hooks';

const VERSION = VEST_VERSION;

export default {
  VERSION,
  create,
  draft,
  enforce,
  group,
  only,
  skip,
  test,
  warn,
};

export { VERSION, create, draft, enforce, group, only, skip, test, warn };
