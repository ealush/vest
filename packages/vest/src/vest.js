import create from 'createSuite';
import enforce from 'enforce';
import { only, skip, warn, group } from 'hooks';
import test from 'test';

const VERSION = __LIB_VERSION__;

export default {
  VERSION,
  create,
  enforce,
  group,
  only,
  skip,
  test,
  warn,
};
