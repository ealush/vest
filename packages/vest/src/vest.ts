import { enforce } from 'n4s';

import create from 'create';
import context from 'ctx';
import each from 'each';
import { only, skip } from 'exclusive';
import group from 'group';
import include from 'include';
import { eager } from 'mode';
import omitWhen from 'omitWhen';
import optional from 'optionalTests';
import skipWhen from 'skipWhen';
import { test } from 'test';
import warn from 'warn';

const VERSION = __LIB_VERSION__;
export {
  test,
  create,
  each,
  only,
  skip,
  warn,
  group,
  optional,
  skipWhen,
  omitWhen,
  enforce,
  VERSION,
  context,
  include,
  eager,
};
