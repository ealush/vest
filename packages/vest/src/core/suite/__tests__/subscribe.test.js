import vest from 'vest';

describe('suite.subscribe', () => {
  let suite;
  let handler;
  beforeEach(() => {
    handler = jest.fn();
    suite = vest.create(() => {});
  });

  it('Should call handler on suite subscription initialization', () => {
    suite();

    expect(handler).toHaveBeenCalledTimes(0);
    suite.subscribe(handler);
    expect(handler).toHaveBeenCalledTimes(1);
  });

  it('Should call handler on suite subscription initialization', () => {
    suite();

    suite.subscribe(handler);
    expect(handler.mock.calls[0]).toMatchSnapshot();
  });
});
