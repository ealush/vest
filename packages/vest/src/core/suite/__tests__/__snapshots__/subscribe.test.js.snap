// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`suite.subscribe Should call handler on suite subscription initialization 1`] = `
Object {
  "suiteState": Object {
    "pending": [Function],
    "suiteId": [Function],
    "testCallbacks": [Function],
    "testObjects": [Function],
  },
  "type": "suiteSubscribeInit",
}
`;
